src/main/java/net/fabricmc/loom/processors/AccessEscalator.java

```diff
@@ -0,0 +1,338 @@
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package net.fabricmc.loom.processors;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.StringWriter;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+
+import org.objectweb.asm.Opcodes;
+
+import net.fabricmc.mappings.EntryTriple;
+
+public class AccessEscalator {
+	public String namespace;
+	public Map<String, Access> classAccess = new HashMap<>();
+	public Map<EntryTriple, ChangeList> methodAccess = new HashMap<>();
+	public Map<EntryTriple, ChangeList> fieldAccess = new HashMap<>();
+
+	public void read(BufferedReader reader) throws IOException {
+		String[] header = reader.readLine().split("\t");
+
+		if (header.length != 2 || !header[0].equals("ae\\v1")) {
+			throw new UnsupportedOperationException("Unsupported or invalid access escalator file");
+		}
+
+		if (namespace != null) {
+			if (!namespace.equals(header[1])) {
+				throw new RuntimeException("Namespace mismatch");
+			}
+		}
+
+		namespace = header[1];
+
+		String line;
+
+		while ((line = reader.readLine()) != null) {
+			//Comment handling
+			int commentPos = line.indexOf('#');
+
+			if (commentPos >= 0) {
+				line = line.substring(0, commentPos).trim();
+			}
+
+			if (line.isEmpty()) continue;
+
+			//Will be a common issue, make it clear.
+			if (line.contains(" ")) {
+				throw new RuntimeException("Access escalator contains spaces, tabs are required on line: " + line);
+			}
+
+			String[] split = line.split("\t");
+
+			if (split.length != 3 && split.length != 5) {
+				throw new RuntimeException("Failed to parse access escalator. at line:" + line);
+			}
+
+			Access access = parseAccess(split[0]);
+
+			switch (split[1]) {
+			case "class":
+				if (split.length != 3) {
+					throw new RuntimeException("Failed to parse access escalator. at line:" + line);
```

Comment:
Instead of aborting immediately, I'd add the encountered errors to a list and continue trying to parse the file, then abort after it's done, that way you can see all the errors at once (and on top of that you don't get a completely irrelevant stack trace except for the message)
I'd also make the error more descriptive, something like
```
syntax error: expected '<class name> <access level>' after 'class'
at /path/to/inputfile.txt:40:
    |
 40 | class sdas as das das d asd as
    |       ^~~~~~~~~~~~~~~~~~~~~~~~
```

Same goes for all the other parse errors.

Reply:
I can easily make the erros more descriptive, im not sure if its worth the effort to go to make it super detailed.

Reply:
Yeah it doesn't have to be this detailed, one line error messages would definitely be enough, the main thing I would want is that it doesn't immediately throw when it encounters the first error.